simexpal/launch/common.py

```diff
@@ -424,65 +424,79 @@ def close(self):
 	start = time.perf_counter()
 	cwd = (util.expand_at_params(manifest.workdir, substitute)
 			if manifest.workdir is not None else manifest.base_dir)
-	child = subprocess.Popen(cmd, cwd=cwd, env=environ,
-			stdout=stdout, stderr=stderr)
-	os.close(stderr)
-	sel = selectors.DefaultSelector()
-
-	if manifest.stdout is None:
-		os.close(stdout)
-		stdout_writer = LazyWriter(stdout_pipe, manifest.aux_file_path('stdout'))
-		sel.register(stdout_pipe, selectors.EVENT_READ, stdout_writer)
-	stderr_writer = LazyWriter(stderr_pipe, manifest.aux_file_path('stderr'))
-	sel.register(stderr_pipe, selectors.EVENT_READ, stderr_writer)
-
-	# Wait until the run program finishes.
-	while True:
-		if child.poll() is not None:
-			break
-
-		elapsed = time.perf_counter() - start
-		if manifest.timeout is not None and elapsed > manifest.timeout:
-			# Programs can catch the SIGXCPU signal and keep running.
-			# Thus, the kill command ensures that the time limit is respected (with a grace period).
-			if elapsed > manifest.timeout + base.TIMEOUT_GRACE_PERIOD:
-				child.kill()
-			else:
-				child.send_signal(signal.SIGXCPU)
-
-		# Consume any output that might be ready.
-		events = sel.select(timeout=1)
-		for (sk, mask) in events:
-			if not sk.data.progress():
-				sel.unregister(sk.fd)
-
-	# Consume all remaining output.
-	while True:
-		events = sel.select(timeout=0)
-		for (sk, mask) in events:
-			if not sk.data.progress():
-				sel.unregister(sk.fd)
-		if not events:
-			break
-	if manifest.stdout is None:
-		stdout_writer.close()
-		os.close(stdout_pipe)
-	stderr_writer.close()
-	os.close(stderr_pipe)
-	runtime = time.perf_counter() - start
-
-	# Collect the status information.
-	status = None
-	sigcode = None
-	if child.returncode < 0:  # Killed by a signal?
-		sigcode = signal.Signals(-child.returncode).name
+	try:
+		child = subprocess.Popen(cmd, cwd=cwd, env=environ,
+				stdout=stdout, stderr=stderr)
+	except FileNotFoundError:
+		import traceback
+
+		# Log the error traceback.
+		with open(manifest.aux_file_path('stderr'), 'w') as f:
+			f.write(traceback.format_exc())
+
+		# Create the .status file.
+		status_dict = {'timeout': False, 'walltime': 0, 'status': 1, 'signal': None}
```

Comment:
Not sure if `status`: 1 is a great way to represent this. Maybe we could set `status` = -1 (which should not be a possible return value for successful executions) and add an attribute such as `error: 'executable_not_found'`?

Reply:
In ``Run._update_status_cache_dict()`` we check for ``status_dict['status'] > 0`` in order to determine failed runs. Also runs that fail without a timeout or signal have ``status_dict['status']== 1``. To use ``status_dict['status']== -1`` we could use the condition ``status_dict['status']> 0 or status_dict['status']< 0`` or alternatively check for the error entry e.g. ``status_dict['status']> 0 or status_dict['error']``. What do you think?

Reply:
I set ``status = -1``  and ``error: 'executable_not_found'`` when the executable is missing and chose the latter approach  i.e. the condition ``status_dict['status']> 0 or status_dict['error']`` to check for this case.